src/choleskyinv.jl

```diff
@@ -0,0 +1,151 @@
+using LinearAlgebra
+
+import Base: show, iterate
+
+"""
+Cholesky factorization and inverse Cholesky factorization, accessible
+in fields `.c` ans `.ci`, respectively.
+"""
+struct Choleskyinv{T<:Factorization}
+    c::T
+	ci::T
+end
+
+# iteration for destructuring into components
+Base.iterate(Choleskyinv) = (C.c, C.ci)

```

Comment:
This doesn't look right as `iterate` should return a current object and a state. You should probably follow the example of QR: https://github.com/JuliaMatrices/MatrixFactorizations.jl/blob/2c95f046f22a694c50d42a7f751a72644633bd42/src/qr.jl#L55

Reply:
I am not sure what to do here. I have removed it for the moment being. Can you please give a try yourself?

Reply:
Actually, just do this:
```julia
Base.iterate(C::CholeskyInv) = interate((C.c,C.ci))
```

Reply:
OK, i am updating. Thanks